src/java.base/share/classes/jdk/internal/crac/mirror/Core.java

```diff
@@ -236,7 +237,27 @@ public String run() {
         }
 
         if (newArguments != null && newArguments.length() > 0) {
-            String[] args = newArguments.split(" ");
+            // Parse arguments into array, unescape spaces
+            final char escChar = '\\';
+            final char sepChar = ' ';
+            ArrayList<String> argList = new ArrayList<String>();
```

Comment:
Could you please rewrite this using `ArrayList<StringBuilder>`?

Reply:
Done